File: 01-create_your_first_lambda_function/node/myLambdaFunction.js
function calculateArea(length, width) {
    return length * width;
}

export const handler = async (event, context) => {
    // Get the length and width parameters from the event object. The
    // runtime converts the event JSON to a JavaScript object.
    const { length, width } = event;

    if (length === undefined || width === undefined) {
        throw new Error("Event must contain 'length' and 'width' keys");
    }

    const area = calculateArea(length, width);

    console.log(`The area is ${area}`);
    console.log('CloudWatch log group:', context.logGroupName);

    return JSON.stringify({ area });
};

File: 01-create_your_first_lambda_function/python/myLambdaFunction.py
import json
import logging

logger = logging.getLogger()
logger.setLevel(logging.INFO)

def lambda_handler(event, context):
    if 'length' not in event or 'width' not in event:
        raise ValueError("Event must contain 'length' and 'width' keys")

    length = event['length']
    width = event['width']

    area = calculate_area(length, width)
    logger.info(f"The area is {area}")
    logger.info(f"CloudWatch logs group: {context.log_group_name}")

    return json.dumps({"area": area})

def calculate_area(length, width):
    return length * width
